addons/account/models/account_reconcile_model.py

```diff
@@ -474,7 +474,11 @@ def _prepare_widget_writeoff_vals(self, st_line_id, write_off_vals):
             'analytic_account_id': counterpart_vals.get('analytic_account_id'),
             'analytic_tag_ids': counterpart_vals.get('analytic_tag_ids', []),
             'reconcile_model_id': self.id,
-            'journal_id': counterpart_vals['journal_id']
+            'journal_id': counterpart_vals['journal_id'],
+            'tax_exigible': counterpart_vals.get('tax_exigible'),
+            'tax_repartition_line_id': counterpart_vals.get('tax_repartition_line_id'),
+            'tax_ids': counterpart_vals.get('tax_ids', []),
+            'tax_tag_ids': counterpart_vals.get('tax_tag_ids', []),
```

Comment:
@smetl I don't understand the changes in https://github.com/odoo/odoo/commit/40bbca9758dcf57f57df0e556eedca203375a7e0 and https://github.com/odoo/enterprise/commit/de6e6afbfcedbf04ca6f712ccf975d77e5c3586d. Why are we not simply returning `counterpart_vals`? (maybe with `reconcile_model_id` added on top of it if it is not already in there)

Reply:
@william-andre I don't remember. Clean it and see if tests are failing or not ?

Reply:
Not exactly. See `'balance': counterpart_vals['amount_currency'],` :S

Reply:
That seems wrong since it means that `balance != debit - credit`...
With a `currency_id` set...